lib/trie/hash.go

```diff
@@ -17,31 +17,64 @@
 package trie
 
 import (
+	"bytes"
+	"context"
 	"hash"
+	"sync"
+
+	"github.com/ChainSafe/gossamer/lib/common"
+	"github.com/ChainSafe/gossamer/lib/scale"
+	"golang.org/x/sync/errgroup"
 
 	"golang.org/x/crypto/blake2b"
 )
 
+type sliceBuffer []byte
+
+func (b *sliceBuffer) Write(data []byte) (n int, err error) {
+	*b = append(*b, data...)
+	return len(data), nil
+}
+
+func (b *sliceBuffer) Reset() {
+	*b = (*b)[:0]
+}
+
 // Hasher is a wrapper around a hash function
 type Hasher struct {
-	hash hash.Hash
+	hash     hash.Hash
+	tmp      sliceBuffer
+	parallel bool // Whether to use parallel threads when hashing
+}
+
+// HasherPool creates a pool of Hasher.
+var HasherPool = sync.Pool{
```

Comment:
same here, probably change this and `Hasher` to unexported

Reply:
Done